### PR TITLE
#6Task. AleksKovalenko

### PR DESCRIPTION
**Pull request by alextrapper**
Successful build is [here](https://travis-ci.org/github/alextrapper/js-assignments/builds/707725337)

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,19 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    if(num % 3 == 0 || num % 5 == 0){
+        let result ="";
+        if(num % 3 == 0) {
+            result += 'Fizz';
+        }
+        if(num % 5 == 0){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator === is always preferable
sometimes ''==' may cause an error
(check at console :
""0"" == 0
and ""0"" === 0)"

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -82,7 +99,12 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if(a < b + c && b < a + c && c < a + b){
+        return true;
+    }
+    else{
+        return false;
+    }
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: We can simply return (a < b + c && b < a + c && c < a + b)
Please, see https://javascript.info/logical-operators

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -166,7 +191,13 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+    str = str.split(' ').join('');
+    for(let i = 0; i < str.length; i++){
+        if(str.indexOf(str[i]) == str.lastIndexOf(str[i])){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -251,7 +284,14 @@ function reverseInteger(num) {
  *   4916123456789012 => false
  */
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+    let reverted = ccn.toString().split('').reverse();
+    let s1 = reverted.reduce((acc, v, i)=>
+        i % 2 == 0 ? acc + +v : acc, 0);
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -251,7 +284,14 @@ function reverseInteger(num) {
...
+    let s2 = reverted.reduce((acc, v, i) =>
+         i % 2 != 0
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
""0"" != 0
and ""0"" !== 0)"

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -396,7 +490,23 @@ function getCommonDirectoryPath(pathes) {
  *
  */
 function getMatrixProduct(m1, m2) {
-    throw new Error('Not implemented');
+    let M_1_Rows = m1.length, M_1_Columns = m1[0].length,
+        M_2_Rows = m2.length, M_2_Columns = m2[0].length;
+    let resultingMatrix;
+
+    if (M_1_Columns != M_2_Rows) return false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator !== is always preferable

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by alextrapper** [comment]
I've made commit with fixes

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**